Conch/lib/Conch/Controller/DeviceValidation.pm

```diff
@@ -21,11 +22,33 @@ Get latest validation states for a device
 =cut
 
 sub list_validation_states ($c) {
+	my @statuses;
+	@statuses = map { lc($_) } split /,\s*/, $c->param('status')
+		if $c->param('status');
+	
+	if (
+		@statuses
+		&& notall {
+			my $a = $_;
+			any { $_ eq $a } qw( pass fail error )
+		}
+		@statuses
+		)
+	{
+		return $c->status(
```

Comment:
Is there no way now to get all validation states? I'd like to retain that ability while also supporting this kind of filtering.

Reply:
If unspecified, all validation states are returned. Corresponding test https://github.com/joyent/conch/pull/243/files#diff-86a28648deb62650c8e4a8eb6d5cf155R461